src/cc/bpf_module_rw_engine.cc

```diff
@@ -50,7 +50,10 @@ void BPFModule::cleanup_rw_engine() {
 static LoadInst *createLoad(IRBuilder<> &B, Value *addr, bool isVolatile = false)
 {
 #if LLVM_MAJOR_VERSION >= 14
-  return B.CreateLoad(addr->getType()->getPointerElementType(), addr, isVolatile);
+  if (isa<AllocaInst>(addr))
+    return B.CreateLoad(dyn_cast<AllocaInst>(addr)->getAllocatedType(), addr, isVolatile);
+  else
+    return B.CreateLoad(addr->getType(), addr, isVolatile);
```

Comment:
I thought this patch is to fix llvm15 issue. Is llvm14 change needed as well?

Reply:
https://github.com/llvm/llvm-project/blob/llvmorg-14.0.0/llvm/include/llvm/IR/Type.h#L369-L373

`getPointerElementType` is deprecated since llvm14. And this change also applies to llvm14.